prep_TS_hyp3.ipynb

```diff
@@ -0,0 +1,500 @@
+    "    geoproj = filehandle.GetProjection()\n",
+    "    return filehandle.RasterXSize, filehandle.RasterYSize, geotransform, geoproj\n",
+    "\n",
+    "def getOrigins(files):\n",
+    "\n",
+    "    ul = np.zeros((2,len(files)))\n",
+    "    lr = np.zeros((2,len(files)))\n",
+    "\n",
+    "    for i in range(len(files)):\n",
+    "        x,y,trans,proj = read_gdal_file_geo(open_gdal_file(files[i]))\n",
+    "        ul[0,i] = trans[0]\n",
+    "        lr[0,i] = trans[0] + x*trans[1]\n",
+    "        ul[1,i] = trans[3]\n",
+    "        lr[1,i] = trans[3] + y*trans[5]\n",
+    "\n",
+    "    return ul,lr,trans[1],trans[5]\n",
+    "\n",
+    "def copyOrigins(files,all_coords,all_pixsize):\n",
+    "    ul = np.zeros((2,len(files)))\n",
+    "    lr = np.zeros((2,len(files)))\n",
+    "\n",
+    "    for i in range(len(files)):\n",
+    "        coords = all_coords[i]\n",
+    "        ul[0,i] = coords[0]\n",
+    "        lr[0,i] = coords[2]\n",
+    "        ul[1,i] = coords[1]\n",
+    "        lr[1,i] = coords[3]\n",
+    " \n",
+    "        if i == 0:\n",
+    "            xres = all_pixsize[i]\n",
+    "            yres = all_pixsize[i]\n",
+    "\n",
+    "    return ul,lr,xres,yres\n",
+    "\n",
+    "def cutGeotiffsByLine(files):\n",
+    "    ul,lr,xres,yres = getOrigins(files)\n",
+    "    \n",
+    "    diff_ul = np.zeros((2,len(files)))\n",
+    "    diff_ul[0] = (max(ul[0])-ul[0])/xres    \n",
+    "    diff_ul[1] = -1*(min(ul[1])-ul[1])/(-1*yres)\n",
+    "\n",
+    "    print(\"Difference list:\")\n",
+    "    print(diff_ul)\n",
+    "\n",
+    "    lrx = min(lr[0])\n",
+    "    lry = max(lr[1])\n",
+    "    lenx = (lrx-max(ul[0])) / xres\n",
+    "    leny = -1*(lry-min(ul[1])) / (-1*yres)\n",
+    "    if leny < 0:\n",
+    "        leny = abs(leny)\n",
+    "        diff_ul[1] = diff_ul[1] * -1\n",
+    "    print(\"Size of output images {} x {}\".format(lenx, leny))\n",
+    "\n",
+    "    outfiles = []   \n",
+    "    for i in range(len(files)):\n",
+    "        outfile = files[i].replace(\".tif\",\"_clip.tif\")\n",
+    "        print(\"Processing file {} to create file {}\".format(files[i], outfile))\n",
+    "        gdal.Translate(outfile,files[i],srcWin=[diff_ul[0,i],diff_ul[1,i],lenx,leny])\n",
+    "        outfiles.append(outfile)\n",
+    "\n",
+    "    return(outfiles)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Clip the files"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get files to be clipped in the outdir and its sub-directories.\n",
+    "files = []\n",
+    "pattern = [\"*.tif\"]                      \n",
+    "for dir,_,_ in os.walk(outdir):\n",
+    "    for item in pattern:\n",
+    "        files.extend(glob.glob(os.path.join(dir, item)))\n",
+    "# clip off files\n",
+    "cutGeotiffsByLine(files)    "
```

Comment:
Generating the list of tif files can be simplified using wildcards:
```
pattern = os.path.join(outdir, '**/*.tif')
files = glob.glob(pattern)
cutGeotiffsByLine(files)
```

Reply:
accept your suggestion